tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -600,6 +600,52 @@ bool containOnlyVecWithFourFloats(QualType type, bool use16Bit) {
   return false;
 }
 
+// Evaluates whether a given QualType is in fact an array and unroll accordingly.
+// Will return the the appropriate RecordType for the provided QualType. Will be 
+// the actual type of the array element, if it's really an array.
+template <size_t SmallVectorSize>
+QualType unrollMultiDimensionalArray(
+    const ASTContext &astContext, const QualType &startType,
+    llvm::SmallVector<int, SmallVectorSize> *arraySizes
+) {
+
+  QualType innerQualType = startType;
+
+  // Unroll a multidimensional array.
+  if (const auto *arrayType = startType->getAsArrayTypeUnsafe()) {
```

Comment:
Use an early exit: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code